src/components/Signup/Signup.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect, useRef } from "react";
+import { useAuth } from "../../context/AuthContext/AuthContext";
+import { navigate } from "@reach/router";
+import axios from "axios";
+import makeApiCall from "../../server/server.request";
+import { validateEmail } from "../../Utilities/validation-utils";
+
+const Signup = () => {
+  const { user, signup, token, setUser } = useAuth();
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
```

Comment:
Instead of  separate states for username,email and password you can have a single state with initial value as an object